tests/testthat/test-basic.R

```diff
@@ -0,0 +1,130 @@
+#### Tests for basic finalsize functionality ####
+
+#### Prepare data ####
+r0 <- 2.0
+polymod <- socialmixr::polymod
+contact_data <- socialmixr::contact_matrix(
+  polymod,
+  countries = "United Kingdom",
+  age.limits = c(0, 20, 40),
+  symmetric = TRUE
+)
+contact_matrix <- contact_data$matrix
+demography_vector <- contact_data$demography$population
+
+# set demography vector names
+names(demography_vector) <- contact_data$demography$age.group
+
+# scale by maximum real eigenvalue and divide by demography
+contact_matrix <- contact_matrix / max(eigen(contact_matrix)$values)
+contact_matrix <- contact_matrix / demography_vector
+
+n_demo_grps <- length(demography_vector)
+n_risk_grps <- 2
+
+# prepare p_susceptibility and susceptibility
+p_susceptibility <- matrix(
+  data = 1, nrow = n_demo_grps, ncol = n_risk_grps
+)
+p_susceptibility <- p_susceptibility / rowSums(p_susceptibility)
+susceptibility <- matrix(
+  data = c(1.0, 0.5), nrow = n_demo_grps, ncol = n_risk_grps,
+  byrow = TRUE
+)
+colnames(susceptibility) <- c("susceptible", "immunised")
+
+# prepare final_size output
+epi_outcome <- final_size(
+  r0 = r0,
+  contact_matrix = contact_matrix,
+  demography_vector = demography_vector,
+  p_susceptibility = p_susceptibility,
+  susceptibility = susceptibility,
+  solver = "iterative"
+)
+
+#### Test that final_size returns a data.frame with correct columns ####
+test_that("final_size output is dataframe with correct names", {
+  # expect data.frame as output
+  expect_s3_class(
+    epi_outcome, "data.frame"
+  )
+
+  # expect that data.frame has correct names
+  expect_identical(
+    colnames(epi_outcome),
+    c("demo_grp", "susc_grp", "susceptibility", "p_infected")
+  )
+
+  # check for names
+  expect_named(
+    demography_vector,
+    unique(epi_outcome$demo_grp)
+  )
+  expect_setequal(
+    epi_outcome$susc_grp, c("susceptible", "immunised")
+  )
```

Comment:
This is tested in `test-check_output_df.R`

Reply:
Thanks; will remove `test-check_output_df.R`, it's an earlier file.